### PR TITLE
feat: [OSM-612] Fix broken tests, restructure internal logic

### PR DESCRIPTION
**Pull request by dotkas**
We've had broken tests for ages, everything was on fire in this repo.

* Dropped a bunch of homebrew'ed stuff in the CircleCI config, instead doing tests in a Dockerfile
* Converted all `tap` tests to `jest`
* Removed some tests that were doing the same thing twice
* Dropped support for very old versions of Node, Pip and Python

No logic changes.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,129 +1,67 @@
...
 jobs:
-  install:
-    <<: *defaults
-    environment:
-      NODE_ENV: develop # Required because base image sets it to 'production'
-    steps:
-      - checkout_and_merge
-      - run:
-          name: Use snyk-main npmjs user
-          command: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" >> .npmrc
```
Comment (dotkas) [inline .circleci/config.yml:73]: It's an open source library 🤷🏻

File: test/test-utils.ts [hunk 2]
```diff
@@ -177,41 +170,6 @@ function pipUninstall(pkgName: string) {
...
-  }
-
-  const origWorkonHome = process.env.WORKON_HOME;
-  process.env.WORKON_HOME = venvsBaseDir;
-
-  return function revert() {
-    process.env.WORKON_HOME = origWorkonHome;
-  };
-}
-
```
Comment (dotkas) [inline test/test-utils.ts:214]: Unused code.

File: test/workspaces/pip-app-with-options/requirements.txt
Comment (dotkas) [inline test/workspaces/pip-app-with-options/requirements.txt:1]: Deprecated by Python. Unmaintained.

**Comment by JamesPatrickGill** [comment]
> Dropped support for very old versions of Node, Pip and Python

Are we still matching CLI

**Review by JamesPatrickGill — APPROVED** [review]

**Pull request merged**